File: src/minecraft/net/machinemuse/api/electricity/TEElectricAdapter.java
package net.machinemuse.api.electricity;

import net.machinemuse.powersuits.common.ModCompatability;
import net.minecraft.item.ItemStack;
import thermalexpansion.api.item.IChargeableItem;

public class TEElectricAdapter extends ElectricAdapter {

	protected ItemStack stack;
	protected IChargeableItem item;

	public TEElectricAdapter(ItemStack stack) {
		this.stack = stack;
		this.item = (IChargeableItem) stack.getItem();
	}

	@Override
	public double getCurrentEnergy() {
		return museEnergyFromMJ(item.getEnergyStored(stack));
	}

	@Override
	public double getMaxEnergy() {
		return museEnergyFromMJ(item.getMaxEnergyStored(stack));
	}

	@Override
	public double drainEnergy(double requested) {
		float requestedMJ = (float) museEnergyToMJ(requested);
		float receivedMJ = item.transferEnergy(stack, requestedMJ, true);
		return museEnergyFromMJ(receivedMJ);
	}

	@Override
	public double giveEnergy(double provided) {
		float providedMJ = (float) museEnergyToMJ(provided);
		float takenMJ = item.receiveEnergy(stack, providedMJ, true);
		return museEnergyFromMJ(takenMJ);
	}

	public static double museEnergyToMJ(double museEnergy) {
		return museEnergy / ModCompatability.getBCRatio();
	}

	public static double museEnergyFromMJ(double mj) {
		return mj * ModCompatability.getBCRatio();
	}
}

File: src/minecraft/net/machinemuse/api/electricity/TEElectricAdapter.scala
package net.machinemuse.api.electricity

import net.minecraft.item.ItemStack
import thermalexpansion.api.item.IChargeableItem
import net.machinemuse.api.electricity.PowerConversions._

class TEElectricAdapter(val stack: ItemStack) extends ElectricAdapter {
  val item = stack.getItem.asInstanceOf[IChargeableItem]

  def getCurrentEnergy: Double = {
    return museEnergyFromMJ(item.getEnergyStored(stack))
  }

  def getMaxEnergy: Double = {
    return museEnergyFromMJ(item.getMaxEnergyStored(stack))
  }

  def drainEnergy(requested: Double): Double = {
    val requestedMJ: Float = museEnergyToMJ(requested).asInstanceOf[Float]
    val receivedMJ: Float = item.transferEnergy(stack, requestedMJ, true)
    return museEnergyFromMJ(receivedMJ)
  }

  def giveEnergy(provided: Double): Double = {
    val providedMJ: Float = museEnergyToMJ(provided).asInstanceOf[Float]
    val takenMJ: Float = item.receiveEnergy(stack, providedMJ, true)
    return museEnergyFromMJ(takenMJ)
  }
}